test/jdk/tools/jpackage/apps/image/Hello.java

```diff
@@ -163,6 +163,12 @@ private static Hello createInstance() {
             ex.printStackTrace();
         }
 
+        if (!Desktop.isDesktopSupported()) {
```

Comment:
Would it make more sense to replace
`if (GraphicsEnvironment.isHeadless())`
with
`if (GraphicsEnvironment.isHeadless() || !Desktop.isDesktopSupported())`
without need to add another `trace()` call?

Reply:
I would prefer to have a separate messages for "headless" and "desktop capable". I can merge these, if you insist.

Reply:
Duplicated messages are not much help in understanding control flow. I'd change the new message or merge conditions. Whatever you prefer.

Reply:
It is fine with me to have two separate messages for display and desktop, but it is better to move this check to top of function after if (GraphicsEnvironment.isHeadless()). No point to execute other code.

Reply:
Okay, I moved the check upwards. The trace message is left intact. Note it says "desktop", there, so it is not a duplicate message.